packages/dara-components/dara/components/smart/chat/chat.py

```diff
@@ -0,0 +1,63 @@
+"""
+Copyright 2023 Impulse Innovations Limited
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from dara.core.definitions import StyledComponentInstance
+from dara.core.interactivity import NonDataVariable
+
+
+class Chat(StyledComponentInstance):
+    """
+    A Chat component which can be added anywhere in your page. When added a chat button will appear in the bottom right
+    of your page and when clicked a chat sidebar will appear. This can be added on a page by page basis, with the chat
+    state being store in a Variable.
+
+    If you would like for the Variable's state to be persistent between restarts, and shared across multiple app users,
+    you can use a BackendStore with your prefered backend. For example, to store the chat messages in a json file, you
+    can use the FileBackend, as showcased below:
```

Comment:
I think this could be made clearer:
1. You *have* to use a BackendStore for this to behave correctly, otherwise each user has their own ephemeral chat state which is quite pointless. Side note: perhaps we should validate the value variable has a BackendStore attached?
2. You can choose to use a persistent backend rather than an ephemeral one (i.e. currently File rather than InMemory) for it to persist across restarts.

Your current wording suggests you have to use a backend store for persistence between restarts but that's not quite correct,s ee above.

Reply:
Updated, let me know if you think this is better, have also added a validation to see if a store is attached